lib/pescarte_web/controllers/relatorio_controller.ex

```diff
@@ -0,0 +1,35 @@
+defmodule PescarteWeb.Pesquisa.RelatorioController do
+  use PescarteWeb, :controller
+
+  alias Pescarte.RelatorioCompiler
+
+  def download_pdf(conn, %{"id" => id}) do
+    pdf_binary = RelatorioCompiler.gerar_pdf(id)
+    enviar_pdf_response(conn, id, pdf_binary)
+  end
```

Comment:
creio que dessa forma fique mais clara a intenção e a possibilidade de retornar um erro

```suggestion
  def download_pdf(conn, %{"id" => id}) do
    with {:ok, pdf_filename, pdf_binary} <- RelatorioCompiler.gerar_pdf(id) do
      conn
      |> put_status(:ok)
      |> send_download({:binary, pdf_binary}, filename: pdf_filename)
    end
  end
```